### PR TITLE
Add docker to run app

### PR DESCRIPTION
**Pull request by bengpotter**
Adds docker files and docker-compose.yml to run app. Not ready to be used as a development environment.

### CODE REVIEW

File: config/database.yml [hunk 1]
```diff
@@ -17,6 +17,9 @@
 default: &default
   adapter: postgresql
   encoding: unicode
+  host: db
+  username: postgres
+  password: penis
```
Comment (AsherWright) [inline config/database.yml]: I don't like having our password in plaintext

File: config/database.yml [hunk 2]
```diff
@@ -82,5 +85,3 @@ test:
 production:
   <<: *default
   database: elderly_grocery_delivery_production
-  username: elderly_grocery_delivery
-  password: <%= ENV['ELDERLY_GROCERY_DELIVERY_DATABASE_PASSWORD'] %>
```
Comment (AsherWright) [inline config/database.yml]: Much prefer keeping this.

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      context: ./Docker
+      dockerfile: db.Dockerfile
+    volumes:
+      - ./tmp/db:/var/lib/postgresql/data
+    expose:
+      - 5432
+    environment:
+      - POSTGRES_HOST_AUTH_METHOD=trust
+    ports:
+      - 5432:5432
```
Comment (AsherWright) [inline docker-compose.yml]: Why 5432?

File: entrypoint.sh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+#!/bin/bash
+set -e
+
+# Remove a potentially pre-existing server.pid for Rails.
+rm -f /elderly-grocery-delivery/tmp/pids/server.pid
```
Comment (AsherWright) [inline entrypoint.sh]: This is hilarious; is this legit?

Comment (bengpotter) [reply entrypoint.sh]: That was in the default dockefile.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,36 @@
+FROM ruby
+
+RUN apt-get update -qq && apt-get install -y nodejs postgresql-client
```
Comment (AsherWright) [inline Dockerfile]: double check nodejs?

File: config/database.yml [hunk 6]
```diff
@@ -82,5 +83,5 @@ test:
 production:
   <<: *default
   database: elderly_grocery_delivery_production
-  username: elderly_grocery_delivery
+  username: benny
```
Comment (AsherWright) [inline config/database.yml]: Shouldn't these be `ENV` too?

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+
+RUN apt-get -y update && \
+      apt-get install --fix-missing --no-install-recommends -qq -y \
+        build-essential \
+        vim \
+        wget gnupg \
+        git-all \
+        curl \
+        ssh \
+        postgresql-client-10 libpq5 libpq-dev -y && \
```
Comment (AsherWright) [inline Dockerfile]: are you using postgresql 10 everywhere?

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+RUN gem install bundler
+#Install gems
+RUN mkdir /gems
+WORKDIR /gems
+COPY Gemfile .
+COPY Gemfile.lock .
+RUN bundle install
+
+ARG INSTALL_PATH=/opt/elderly
```
Comment (AsherWright) [inline Dockerfile]: path is `elderly`?

Comment (bengpotter) [reply Dockerfile]: Inside the container the app lives in /opt/elderly

File: config/database.yml (hunk 6 above)
Comment (bengpotter) [reply config/database.yml]: yes.

File: Dockerfile (hunk 7 above)
Comment (bengpotter) [reply Dockerfile]: updated to 12.

File: Dockerfile (hunk 8 above)
Comment (bengpotter) [reply Dockerfile]: inside the container the app lives in /opt/eldery

**Pull request merged**